tests/gis_tests/test_postgis_extension.py

```diff
@@ -0,0 +1,105 @@
+import unittest
+
+from django.db import connection
+from django.test import TestCase
+
+
+@unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific tests")
+class TestsPostGISCreateExtension(TestCase):
+
+    def setUp(self):
+
+        """make sure the extension is not available initially"""
+
+        supported = [
+            "2.1",
+            "2.2",
+            "2.3",
+            "2.4",
+            "2.4.1"
+        ]
+
+        new_connection = connection.copy()
+
+        try:
+            with new_connection.cursor() as cursor:
+
+                cursor.execute(
+                    """
+                    SELECT installed_version FROM
+                    pg_available_extensions WHERE name = 'postgis';
+                    """
+                )
+
+                if cursor.fetchone()[0] in supported:
+                    cursor.execute("""DROP EXTENSION postgis CASCADE;""")
+                else:
+                    raise("Unsupported version of PostGIS")
+        finally:
+            new_connection.close()
+
+    def test_extension_is_created(self):
+
+        """test that the PostGIS extension is created if it does not exists"""
+
+        new_connection = connection.copy()
+
+        try:
+            with new_connection.cursor() as cursor:
+
+                cursor.execute(
+                    """SELECT installed_version
+                    FROM pg_available_extensions
+                    WHERE name ='postgis';"""
+                )
+
+                self.assertIsNone(cursor.fetchone()[0])
+
+                new_connection.prepare_database()
+
+                cursor.execute(
+                    """SELECT installed_version
+                    FROM pg_available_extensions
+                    WHERE name ='postgis';"""
+                )
+
+                self.assertTrue(cursor.fetchone()[0])
+                new_connection.close()
+        finally:
+            new_connection.close()
+
+    def test_extension_is_not_created(self):
+
+        """more or less the inverse of the above test"""
+
+        new_connection = connection.copy()
+
+        try:
+            with new_connection.cursor() as cursor:
+
+                cursor.execute(
+                    """SELECT installed_version
+                    FROM pg_available_extensions
+                    WHERE name ='postgis';"""
+                )
+
+                if cursor.fetchone()[0] is None:
+                    try:
+                        new_connection.prepare_database()
+                    except connection.ProgrammingError as e:
+                        self.assertEqual(
```

Comment:
This won't happen because the tests couldn't start running on PostGIS.

Reply:
Removed in next push.